ui/app/src/components/ItemDisplay/ItemDisplay.tsx

```diff
@@ -351,6 +360,16 @@ const ItemDisplay = forwardRef<HTMLSpanElement, ItemDisplayProps>((props, ref) =
         title={item.label}
         children={item.label}
       />
+      {showPreviewUrl && (
+        <Typography
+          noWrap
+          color="textSecondary"
+          {...pathTypographyProps}
+          className={clsx(classes.path, labelTypographyProps?.className)}
+          title={item.path}
+          children={`• ${item.previewUrl}`}
+        />
+      )}
```

Comment:
This is too wired for something too specific. Show this in place where needed. ItemDisplay shouldn't change at all.